### PR TITLE
cylance,ti_opencti: fix up package validation issues

### PR DESCRIPTION
**Pull request by efd6**
## Proposed commit message

```
cylance: fix kibana.version syntax in manifest
ti_opencti: use extended list of expected_values for indicator type
```

## Checklist

- [ ] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [ ] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [ ] 

## Related issues

-

### CODE REVIEW

File: packages/ti_opencti/changelog.yml [hunk 1]
```diff
@@ -1,4 +1,9 @@
 # newer versions go on top
+- version: "3.0.0"
```
Comment (efd6) [inline packages/ti_opencti/changelog.yml:2]: Breaking because `threat.indicator.type` values changed.

**Review by andrewkroh — COMMENTED** [review]
LGTM. But we should wait for @chrisberkhout to take look.

**Comment by efd6** [comment]
Agreed. That's why I did not include this in the previous batch. I'm not wildly satisfied with the available labels.

**Comment by chrisberkhout** [comment]
The expected values issue seems to keep coming up.

### Regarding the validation of expected values

When I created the OpenCTI integration I included pipeline test examples for each type of indicator that I had data for, which included some that weren't in the `expected_values` list for `threat.indicator.type`, so `elastic-package` failed the build.

When the `expected_values` config option was added it wasn't intended to be a list of the only possible values. The [available documentation](https://github.com/elastic/ecs/blob/4b2c7c6c4596b8d4acdd420907952a04e20c8c23/schemas/README.md?plain=1#L169) still says:

> expected_values (optional): An array of expected values for the field. Schema consumers can validate integrations and mapped data against the listed values. These values are the recommended convention, but users may also use other values.

There's also an [`allowed_values`](https://github.com/elastic/ecs/blob/4b2c7c6c4596b8d4acdd420907952a04e20c8c23/schemas/README.md?plain=1#L187-L189) config option that is different.

I raised the issue in https://github.com/elastic/elastic-package/issues/1472 and it was decided that when we have expected values, we'll continue to enforce those as the only valid values, but we'll allow integrations to override those settings with a different list of expected values (implemented in https://github.com/elastic/package-spec/pull/616).

I had different variations of the config that got the build to pass at different times. The [last version](https://github.com/elastic/integrations/blob/10287a4b7df15550bd2a39c6937d11219dd5f227/packages/ti_opencti/data_stream/indicator/fields/ecs.yml#L1-L9) overrode the definition of `threat.indicator.type`, but without providing a list of expected values. I guess that stopped working? The best update might be to add a list of `expected_values` to that definition.

### Regarding the expected values themselves

The [ECS documentation of `threat.indicator.type`](https://www.elastic.co/guide/en/ecs/current/ecs-threat.html#field-threat-indicator-type) describes it as:

> Type of indicator as represented by Cyber Observable in STIX 2.0.

However, the list of expected values misses some types from STIX and adds some others. OpenCTI uses STIX 2.1 type names but adds several extras. In colums D-G of the ["OpenCTI observable types" tab of the "OpenCTI Mappings" spreadsheet](https://docs.google.com/spreadsheets/d/1w5y_2XVWBZCFaMFPiFQ_jehjsH9Nntfwa8a3nKuxZCs/edit#gid=1509835836), I made a summary of all the type names and which lists they are in.

I'm not sure if alert rules or Security UI features depend on having specific `threat.indicator.type` values, but if so an unknown extra value might still be better than changing to the closest type in the expected list.

The STIX standard has a fixed list of types and no defined way to mark an extra type or unknown type. 

I think the ECS documentation should be updated to say `threat.indicator.type` uses names from STIX 2.1 where possible, but allows other values. The list of expected values could be updated and remain, or it could be changed to a list of example values. If necessary the OpenCTI integration can override the definition with an expanded list. I do think we should keep the expanded list for OpenCTI. However, if we did remove values from `threat.indicator.type`, we should keep the original value in a new field `opencti.indicator.type` and update [a README section](https://github.com/elastic/integrations/blob/main/packages/ti_opencti/_dev/build/docs/README.md#logs).

**Review by chrisberkhout — APPROVED** [review]
Added a couple of commits.

**Comment by efd6** [comment]
Thanks

**Pull request merged**